platform/openide.explorer/src/org/openide/explorer/view/NodeTableModel.java

```diff
@@ -218,9 +218,9 @@ private void computeVisiblePorperties(int visCount) {
             int vi = allPropertyColumns[i].getVisibleIndex();
 
             if (vi == -1) {
-                sort.put(new Double(i - 0.1), Integer.valueOf(i));
+                sort.put(Double.valueOf(i - 0.1), Integer.valueOf(i));
             } else {
-                sort.put(new Double(vi), Integer.valueOf(i));
+                sort.put(Double.valueOf(vi), Integer.valueOf(i));
```

Comment:
I would assume, that autoboxing should do the right thing. In line 221 both `#valueOf` calls can be dropped and in line 223 the second `#valueOf` can be dropped and the first can be replaced with a direct cast.

Reply:
I looked at the JLS and I'm going to leave it as is if you are okay with this.

The reason is is the JLS section 5.1.3 Narrowing Primitive Conversions talks about how valueOf() gets handled. The JLS section 5.5. Casting Contexts handles type casting and it's not really clear both narrow/expand the cast the same way. I would really have to look at this in great detail and the work just didn't justify the brain power. 8-)

So I decided to just leave it alone as this guarantees no change in behavior.

Reply:
Final thought: please compare to `ide/bugtracking.commons/src/org/netbeans/modules/bugtracking/issuetable/NodeTableModel.java`. The logic looks identical, aligning the code would be logical IMHO.